packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/editor/inputs/DataSourceComboBox.kt

```diff
@@ -98,14 +100,9 @@ class DataSourceComboBox(
         var isFirstInit = true
         coroutineScope.launch {
             project.getToolbarModel().toolbarState.collect { state ->
-                withoutSelectionChangedListener {
-                    selectedDataSourceConnecting = state.selectedDataSourceConnecting
-                    selectedDataSourceFailedConnecting = state.selectedDataSourceConnectionFailed
-                    if (isFirstInit || state.dataSources != dataSources) {
-                        isFirstInit = false
-                        populateComboBoxWithDataSources(state.dataSources)
-                    }
-                    selectDataSourceByUniqueId(state.selectedDataSource?.uniqueId)
+                SwingUtilities.invokeLater {
```

Comment:
Why are we using the SwingUtilities invokeLater and not the Application invokeLater?

Reply:
This is what my IDE autocompleted when I used invokeLater. I now see that in every other place we use Application.invokeLater, will switch to that but just for my clairification - are these two same or not?

Reply:
Application.invokeLater does a few optimisations depending on the thread you are one and it's IntelliJ specific, so we should do that unless we have a really strong reason to avoid it.

Reply:
No reason other than autocomplete - I will make the change. Thank you for the suggestion 🙌